games-action/dxx-rebirth/dxx-rebirth-0.61.0_pre20190720.ebuild

```diff
@@ -0,0 +1,245 @@
+# feature.  This variable is stored here to be ready to activate when
+# Portage implements this feature.
+#
+# Note that while individual music formats can be switched without a
+# rebuild, the master flag USE=music controls whether sdl-mixer is used,
+# and therefore requires a rebuild when changed.
+#IUSE_RUNTIME="flac l10n_de midi mp3 opl3-musicpack sc55-musicpack vorbis"
+
+dxx_scons() {
+	# Always build profile `m`.  If use editor, also build profile `e`.
+	# Set most variables in the default anonymous profile.  Only
+	# `builddir` and `editor` are set in the named profiles, since those
+	# must be different between the two builds.
+	#
+	# Notes for end users:
+	# - Per-engine options can be set with an engine prefix, as shown
+	#   for sharepath.  Such options are used in place of
+	#   engine-unqualified options.  For example, to disable sdlmixer
+	#   for Descent 2: Rebirth, but use the setting below for Descent 1:
+	#   Rebirth, set EXTRA_ESCONS="d2x_sdlmixer=0".
+	# - Profile `site` is searched before the anonymous profile, and is
+	#   deliberately unused in this ebuild.  Set options in the site
+	#   profile to affect both engines:
+	#   EXTRA_ESCONS="site_verbosebuild=0".
+	local scons_build_profile=m mysconsargs=(
+		sdl2=$(usex sdl2 1 0)
+		sdlmixer=$(usex music 1 0)
+		verbosebuild=1
+		debug=$(usex debug 1 0)
+		ipv6=$(usex ipv6 1 0)
+		opengl=$(usex opengl 1 0)
+		use_tracker=$(usex tracker 1 0)
+		prefix="${EPREFIX}"/usr
+		screenshot=$(usex png png legacy)
+		m_builddir=build/main/
+		m_editor=0
+	)
+	if use editor; then
+		scons_build_profile+=+e
+		mysconsargs+=( \
+			e_builddir=build/editor/
+			e_editor=1
+		)
+	fi
+	if ! use joystick; then
+		mysconsargs+=( \
+			max_joysticks=0
+		)
+	fi
+	# Add sharepath and enable build of selected games.  The trailing
+	# comma after `$scons_build_profile` is required to cause scons to
+	# search the anonymous profile.  If omitted, only settings from the
+	# named profile would be used.
+	use d1x && mysconsargs+=( d1x_sharepath="/usr/share/games/d1x" d1x="$scons_build_profile,site," )
+	use d2x && mysconsargs+=( d2x_sharepath="/usr/share/games/d2x" d2x="$scons_build_profile,site," )
+	use valgrind && mysconsargs+=( poison=valgrind )
+	escons "${mysconsargs[@]}" "$@"
+}
+
+src_compile() {
+	tc-export CXX PKG_CONFIG
+	dxx_scons register_install_target=0 build
+}
+
+src_install() {
+	# Use upstream install target to handle the various combinations of
+	# enabled/disabled engines and optional editor support.
+	dxx_scons register_compile_target=0 register_install_target=1 DESTDIR="$D" "$D"
+	local DV
+	for DV in 1 2; do
+		if ! use d${DV}x; then
+			continue
+		fi
+		local PROGRAM=d${DV}x-rebirth
+		docinto "${PROGRAM}"
+		dodoc "${PROGRAM}"/*.txt
+		make_desktop_entry "${PROGRAM}" "Descent ${DV} Rebirth" "${PROGRAM}"
+		doicon "${PROGRAM}/${PROGRAM}.xpm"
+	done
+}
+
+pkg_postinst() {
+	if ! use data; then
+		xdg_pkg_postinst
```

Comment:
Why is `xdg_pkg_postinst` inside the `if ! use data` block?  Do users who install data not need this call?  The function looks like it would be appropriate any time the ebuild provides desktop files, icon files, or mime info files.  Since USE=data only controls dependencies, I think the ebuild would either always need this or never need this, but the need would not be conditional on USE=data.

Reply:
Oops, absolutely my error in putting the replacement for the call to 'default' on the wrong line. Will correct this.